.github/workflows/deploy-preview.yml

```
@@ -2,19 +2,94 @@ name: Deploy Preview to Staging
+          isort --check-only .
+          flake8 . --max-line-length=88 --extend-ignore=E203,W503
+          pytest -x
 
   deploy-backend:
     name: Deploy Backend Preview to Railway Staging
-    needs: test
+    needs: test-backend
     runs-on: ubuntu-latest
+    if: |
+      success() &&
+      github.event_name == 'push' &&
```

actionlint (1.7.7)

context "secrets" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)